core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -914,17 +934,18 @@ class CtxSettings : public SettingsBase {
         return ret; // N.B. returns those <key:value> parameters that could not be set
     }
 
-    void addStoredParameters(const property_map& newParameters, const SettingsCtx& ctx) {
+    NO_INLINE void addStoredParameters(const property_map& newParameters, const SettingsCtx& ctx) {
         if (!_autoUpdateParameters.contains(ctx)) {
             _autoUpdateParameters[ctx] = getBestMatchAutoUpdateParameters(ctx).value_or(_allWritableMembers);
         }
 
-        _storedParameters[ctx.context].push_back({ctx, newParameters});
-        auto& vec = _storedParameters[ctx.context];
-        std::sort(vec.begin(), vec.end(), [](const auto& a, const auto& b) { return a.first.time < b.first.time; });
+        std::vector<std::pair<SettingsCtx, property_map>>& sortedVectorForContext = _storedParameters[ctx.context];
+        // binary search and merge-sort
+        auto it = std::ranges::lower_bound(sortedVectorForContext, ctx.time, std::less<>{}, [](const auto& pair) { return pair.first.time; });
+        sortedVectorForContext.insert(it, {ctx, newParameters});
```

Comment:
`std::vector::insert` can be quite costly, so using `std::sort` might be a better option in this case. However, without concrete benchmark measurements, it's difficult to make a definitive judgment, and I don't have a strong preference either way.

Reply:
The cost for this insert sort is lower than a regular sort because the list is already sorted. I.e. in most cases the field is added to the end. In the other it's one insert followed by a linear shift ( `O = N`) ... much faster than any other `O = N log N` sorts